guides/keras_nnx_guide.py

```diff
@@ -0,0 +1,304 @@
+"""
+Title: How to use Keras with NNX backend
+Author: [Divyashree Sreepathihalli](https://github.com/divyashreepathihalli)
+Date created: 2025/08/07
+Last modified: 2025/08/07
+Description: How to use Keras with NNX backend
+Accelerator: CPU
+"""
+
+"""
+
+# A Guide to the Keras & Flax NNX Integration
+
+This tutorial will guide you through the integration of Keras with Flax's NNX
+(Neural Networks JAX) module system, demonstrating how it significantly
+enhances variable handling and opens up advanced training capabilities within
+the JAX ecosystem. Whether you love the simplicity of model.fit() or the
+fine-grained control of a custom training loop, this integration lets you have
+the best of both worlds. Let's dive in!
+
+# Why Keras and NNX Integration?
+
+Keras is known for its user-friendliness and high-level API, making deep
+learning accessible. JAX, on the other hand, provides high-performance
+numerical computation, especially suited for machine learning research due to
+its JIT compilation and automatic differentiation capabilities. NNX is Flax's
+functional module system built on JAX, offering explicit state management and
+powerful functional programming paradigms
+
+NNX is designed for simplicity. It is characterized by its Pythonic approach,
+where modules are standard Python classes, promoting ease of use and
+familiarity. NNX prioritizes user-friendliness and offers fine-grained control
+over JAX transformations through typed Variable collections.
+
+The integration of Keras with NNX allows you to leverage the best of both
+worlds: the simplicity and modularity of Keras for model construction,
+combined with the power and explicit control of NNX and JAX for variable
+management and sophisticated training loops.
+
+# Getting Started: Setting Up Your Environment
+```
+pip install -q -U keras
```

Comment:
Why not actual `!pip install ...`. I think this is needed to run the latest on colab.